packages/ui/src/contexts/ToastContext.tsx

```diff
@@ -1,78 +1,55 @@
-import React, { useCallback, useRef, useState } from 'react'
-import ToastBar from '../components/ToastBar'
-import { Toast } from '../components/ToastContent'
-
-type ToastContextProps = {
-  children: React.ReactNode | React.ReactNode[]
+import { useState, useContext, createContext } from 'react'
+import { ToastType } from '../components/ToastContent'
+import Snackbar from '../components/Snackbar'
+
+export type Toast = {
+  id: number
+  title?: string
+  link?: string
+  duration?: number
+  type: ToastType
 }
 
-export interface IToastContext {
-  addToast: (toastParams: Omit<Toast, 'id'>) => number
-  removeToast: (toastId: number) => void
-  removeAllToasts: () => void
+const MAX_VISIBLE_TOASTS = 6
+
+export type ToastContextProps = {
   toasts: Toast[]
+  setToasts: (toasts: Toast[]) => void
+  addToast: (toast: Omit<Toast, 'id'>) => void
+  removeToast: (id: Toast['id']) => void
 }
 
-const ToastContext = React.createContext<IToastContext | undefined>(undefined)
+const ToastContext = createContext<ToastContextProps | undefined>(undefined)
 
-const ToastContextProvider = ({ children }: ToastContextProps) => {
-  const [toastQueue, setToastQueue] = useState<Toast[]>([])
-  // using useRef instead of useState to keep a tracker over the exact toast array
-  const toasts = useRef<Toast[]>([])
+const ToastProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
+  const [toasts, setToasts] = useState<Toast[]>([])
```

Comment:
as an alternative to not handling the state on our side, we can use https://www.npmjs.com/package/react-toastify
But it's up to us if we want to support additional code

Reply:
I think it's simple enough for us to handle this directly for now at least.